sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -114,19 +115,32 @@ class AttributeMap
         });
   }
 
+  // Construct map from initializer list by applying `SetAttribute` transform for every attribute
```

Comment:
@lalitb - I switched Resource Attributes to use this `AttributeMap` class we had in common utils. I also added this method here that safely transforms from non-owning API/ABI stable collection of `common::AttributeValue` to owning `OwnedAttributeValue`, using `nostd::visit` converter that ensures that for each attribute - we store these as `std::string`.

I removed `const char *` from `OwnedAttributeValue` itself. Fixes #817 , as discussed during community meeting.